### ISSUE TITLE
Level not loading properly from editor

### ISSUE THREAD

**Issue by breadloaf64** [breadloaf64/Chime#2]
Start the game at the titlescreen. navigate through level editor/new level and make some edits to the level. Press enter to playtest, but the level does not load.
The strange thing is that the level has no problem loading when starting from the level editor.

**Comment by breadloaf64** [comment]
<p>I think it's a problem with the singleton pattern implementation for the SessionController, which stores the LevelObject to be loaded. To test this I've added an int counter, which ticks each time GetLevel() or SetLevel() is called and outputs the value. If the SessionController is indeed a singleton that persists through the changing of scenes, the debug log should simply count upwards.</p>
Starting the game from LevelEditor and pressing enter to load the level scene, the debug output looks like this: <br>

![image](https://user-images.githubusercontent.com/60357301/127403960-37425712-268a-474a-87b5-7404617ad0cc.png)

<br>
Starting the game from LevelEditorOptions, selecting new level and pressing enter to load the level scene, the debug output looks like this: <br>

![image](https://user-images.githubusercontent.com/60357301/127403819-269afabe-74e2-4e94-aaac-9acbb7b5ae47.png)

Clearly something fishy is afoot...

**Comment by breadloaf64** [comment]
For extra insight, I got it to print the current scene in each debug.log too: <br>
From LevelEditor:

![image](https://user-images.githubusercontent.com/60357301/127404674-4e8a8694-edc4-45f9-ace1-166289809122.png)

From LevelEditorOptions:

![image](https://user-images.githubusercontent.com/60357301/127404737-f84be951-ca6a-409c-8e25-b1f5caf365f9.png)

How the hell can the counter tick downwards by one in that last log? Like, wherever there is a counter++, it is immediately followed by a debug.log(), thus there's no way for the counter to tick without it being logged... I'm tearing my hair out. I'd better go to bed.

**Comment by breadloaf64** [comment]
It's a problem with the singleton pattern. In the awake method of SessionController, I set the name of that instance. If it the only instance, it is called "the one" and if there are other instances, it is called "bert". In the get and set methods, I print the name. <br>
This turned out to be quite revealing:

![image](https://user-images.githubusercontent.com/60357301/127564543-9d43e36f-a58c-44b5-bf45-d5d26220caba.png)

In that run, we navigate from the title screen to LevelEditorOptions to LevelEditor to Level. It looks as though in LevelEditor, the second instance (bert) does not get destroyed for some reason? But clearly, as soon as the instance is named bert, it should be destroyed:

![image](https://user-images.githubusercontent.com/60357301/127564847-f160d91e-130a-45f4-bc39-b7bf2d5d94ec.png)

**Comment by breadloaf64** [comment]
Clearly the implementation of singleton was not working. This [forum post](https://answers.unity.com/questions/431271/destroy-object-before-end-of-awake-function.html) seems to indicate that `Destroy(gameObject)` doesn't destroy the gameObject immediately which is perhaps why the second instance of SessionController was persisting. I don't fully get it.<br>

Anyway, I implemented [this method](https://gamedev.stackexchange.com/questions/116009/in-unity-how-do-i-correctly-implement-the-singleton-pattern) (top answer) which seemed to work (after adding `Don'tDestroyOnLoad(gameObject)`). Then I replaced every `FindObjectOfType<SessionController>()` with simply:` SessionController.Instance`. Here's how the implementation looks now:

![image](https://user-images.githubusercontent.com/60357301/127569854-8b436b0f-93a8-415c-8cac-4ae373e145f1.png)
